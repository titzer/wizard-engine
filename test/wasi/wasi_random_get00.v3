def buf = Array<byte>.new(16);
def expected: Array<byte> = [
	0xD8, 0x65, 0x88, 0x3C, 0xC5, 0xB5, 0x08, 0x9A, 0xAE, 0xEB, 0x89, 0x47, 0xB6, 0xE9, 0x55, 0xD6
];

def hexMap_u = "0123456789ABCDEF";
def chbuf: Array<byte> = ['0', 'x', 0, 0, ',', ' '];
def iovecs = Array<(Pointer, int)>.new(1);
def sizebuf = Array<u64>.new(1);

def main() -> int {
	var result = wasi_snapshot_preview1.random_get(Pointer.atContents(buf), buf.length);
	if (result != 0) return 1;
	for (b in buf) {
		chbuf[2] = hexMap_u[u4.view(b >> 4)];
		chbuf[3] = hexMap_u[u4.view(b)];
		iovecs[0] = (Pointer.atContents(chbuf), chbuf.length);
		wasi_snapshot_preview1.fd_write(1, Pointer.atContents(iovecs), 1, Pointer.atContents(sizebuf));
	}

	for (i < expected.length) {
		if (buf[i] != expected[i]) return 2 + i;
	}
	return 0;
}
