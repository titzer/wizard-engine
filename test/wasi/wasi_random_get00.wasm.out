0xD8, 0x65, 0x88, 0x3C, 0xC5, 0xB5, 0x08, 0x9A, 0xAE, 0xEB, 0x89, 0x47, 0xB6, 0xE9, 0x55, 0xD6, 